index.bs

```diff
@@ -97,15 +103,109 @@ Introduction {#intro}
 
 <section class="non-normative">
 
-This module describes a mechanism for allowing WebXR applications to cast rays into the users real world environment and report back, to the best of the XR device's knowledge, the point at which the ray intersected with a physical object and the orientation of the intersected surface. This allows for virtual objects to be placed in alignment with those surfaces,
-such as placing objects realistically on the floor or attaching them to a wall.
+This potential module describes a mechanism for allowing WebXR applications to cast rays into the users real world environment and report back, to the best of the XR device's knowledge, the point at which the ray intersected with a physical object and the orientation of the intersected surface. This allows for virtual objects to be placed in alignment with those surfaces, such as placing objects realistically on the floor or attaching them to a wall.
 
 </section>
 
 WebXR Device API Integration {#webxr-device-api-integration}
 ============================
 This module expands the definition of {{XRSession}}.
 
+Geometric Primitives {#geometric-primitives}
+====================
+
+XRRay {#xrray-interface}
+-----
+
+ISSUE: Find the correct spec home for this type
+
+An {{XRRay}} is a geometric ray described by an {{XRRay/origin}} point and {{XRRay/direction}} vector.
+
+An {{XRRay}} contains an <dfn for=XRRay>internal matrix</dfn> which is a [=/matrix=].
+
+<pre class="idl">
+[SecureContext, Exposed=Window,
+ Constructor(optional DOMPointInit origin, optional DOMPointInit direction),
+ Constructor(XRRigidTransform transform)]
+interface XRRay {
+  [SameObject] readonly attribute DOMPointReadOnly origin;
+  [SameObject] readonly attribute DOMPointReadOnly direction;
+  readonly attribute Float32Array matrix;
```

Comment:
Semi-unrelated nit: This should be [SameObject] as well. Feel free to ignore till a future PR if desired.